src/Orleans/Core/GrainFactory.cs

```diff
@@ -85,6 +97,182 @@ public static TGrainInterface GetGrain<TGrainInterface>(string primaryKey, strin
                     typeof(TGrainInterface)));
         }
 
+        /// <summary>
+        /// Creates a reference to the provided <paramref name="obj"/>.
+        /// </summary>
+        /// <typeparam name="TGrainObserverInterface">
+        /// The specific <see cref="IGrainObserver"/> type of <paramref name="obj"/>.
+        /// </typeparam>
+        /// <param name="obj">The object to create a reference to.</param>
+        /// <returns>The reference to <paramref name="obj"/>.</returns>
+        public static Task<TGrainObserverInterface> CreateObjectReference<TGrainObserverInterface>(IGrainObserver obj)
```

Comment:
The problem with that approach is that the specific type of `obj` will become `TGrainObserverInterface`, whereas it should be the interface.
See this comment: https://github.com/dotnet/orleans/issues/59#issuecomment-72548951